SMPyBandits/Policies/FEWA.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+""" The Filtering on Expanding Window Average policy for rotting bandits.
+Reference: [Seznec et al.,  2018].
+"""
+from __future__ import division, print_function  # Python 2 compatibility
+
+__author__ = "Julien Seznec"
+__version__ = "0.1"
+
+from math import sqrt, log
+import numpy as np
+import pandas as pd
+np.seterr(divide='ignore')  # XXX dangerous in general, controlled here!
+
+try:
+    from .BasePolicy import BasePolicy
+except ImportError:
+    from BasePolicy import BasePolicy
+
+
+class EFF_FEWA(BasePolicy):
+    """
+    Efficient mechanism as described in [Seznec et al.,  2019a] (m=2) and [Seznec et al.,  2019b] (m<=2)
+    Algorithm Filtering on Expanding Window Average [Seznec et al.,  2019a]
+    We use the confidence level \delta_t = \frac{1}{t^\alpha}.
+     """
+    def __init__(self, nbArms, alpha=0.06, subgaussian=1, m=2):
+        super(EFF_FEWA, self).__init__(nbArms)
+        self.alpha = alpha
+        self.nbArms = nbArms
+        self.subgaussian = subgaussian
+        self.statistics = np.ones(shape=(3, self.nbArms, 2)) * np.nan
+        # [0,:,:] : current statistics, [1,:,:]: pending statistics, [3,:,:]: number of sample in the pending statistics
+        self.windows = np.array([1, int(np.ceil(m))])
+        self.outlogconst = np.sqrt(self.windows * sqrt(8 * self.alpha * self.subgaussian ** 2))
+        self.armSet = np.arange(nbArms)
+        self.grid = m
+
```

Comment:
There is a `startGame()` method which is missing, it should reset all the internal memories (these arrays) to their default value.
It's not necessary for basic simulations, but all the "wrapper" algorithms (e.g., doubling trick) are using `policy.startGame()` to reset the algorithm (instead of recreating one).